workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs

```diff
@@ -69,65 +58,11 @@ protected override void OnUpdate()
             }
         }
 
-        private void AddEntity(EntityId entityId)
-        {
-            if (worker.EntityIdToEntity.ContainsKey(entityId))
-            {
-                throw new InvalidSpatialEntityStateException(
-                    string.Format(Errors.EntityAlreadyExistsError, entityId.Id));
-            }
-
-            Profiler.BeginSample("OnAddEntity");
-
-            var entity = EntityManager.CreateEntity();
-
-            EntityManager.AddComponentData(entity, new SpatialEntityId
-            {
-                EntityId = entityId
-            });
-
-            EntityManager.AddComponent(entity, ComponentType.Create<NewlyAddedSpatialOSEntity>());
-
-            worker.EntityIdToEntity.Add(entityId, entity);
-            Profiler.EndSample();
-        }
-
-        private void RemoveEntity(EntityId entityId)
-        {
-            if (!worker.TryGetEntity(entityId, out var entity))
-            {
-                throw new InvalidSpatialEntityStateException(
-                    string.Format(Errors.EntityNotFoundForDeleteError, entityId.Id));
-            }
-
-            Profiler.BeginSample("OnRemoveEntity");
-            EntityManager.DestroyEntity(worker.EntityIdToEntity[entityId]);
-            worker.EntityIdToEntity.Remove(entityId);
-            Profiler.EndSample();
-        }
 
         private void OnDisconnect(string reason)
         {
             EntityManager.AddSharedComponentData(worker.WorkerEntity,
```

Comment:
It feels like this belongs in the `EcsViewSystem`

Reply:
Oops missed that. It just seemed reasonable to me last night the the worker would do something on disconnect. We should also make the disconnect event work outside the ecs.